src/main/java/jenkins/plugins/office365connector/utils/FormUtils.java

```diff
@@ -17,17 +17,21 @@
      * @param value user input to validate
      * @return OK or Error on pending user input
      */
-    public static FormValidation formValidateUrl(String value) {
+    public static FormValidation formValidateUrl(String value, String urlCredentialsId) {
+        if (StringUtils.isBlank(value)) {
+            if (!StringUtils.isBlank(urlCredentialsId)) {
+                return FormValidation.ok();
+            } else {
+                return FormValidation.error("Either URL or URL Credentials must be non-empty");
```

Comment:
anyway, did you try to add support for credentials using plugin which I mentioned earlier?

if so the only IMO thing which should be changed is https://github.com/jenkinsci/office-365-connector-plugin/pull/102/files#diff-bd70be8a13679d8af35645c64df15ceaR7 from textbox to password

Reply:
It's supposed to contain the id of a credential, does that need to be hidden using `password`?